hexa/databases/api.py

```diff
@@ -0,0 +1,96 @@
+import re
+
+import psycopg2
+from django.conf import settings
+from django.core.exceptions import ValidationError
+from psycopg2 import sql
+from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
+
+
+def get_db_server_credentials():
+    return {
+        "role": settings.WORKSPACE_DATABASE_ROLE,
+        "password": settings.WORKSPACE_DATABASE_PASSWORD,
+        "host": settings.WORKSPACE_DATABASE_HOST,
+        "port": settings.WORKSPACE_DATABASE_PORT,
+    }
+
+
+def format_db_name(name: str):
+    db_name = ""
+    for char in name.lower().replace(" ", "_"):
+        if char in "abcdefghijklmnopqrstuvwxyz0123456789_":
+            db_name += char
+
+    # Ensure db_name do not start with a number
+    if db_name[0] in "0123456789":
+        db_name = db_name.replace(db_name[0], "_", 1)
+
+    return db_name[:31]
+
+
+def validate_db_name(name: str):
+    if not name:
+        raise ValidationError("Empty value for name")
+
+    if (re.match(r"^[_a-z][a-z0-9_]{,30}$", name)) is None:
+        raise ValidationError(
+            "Name must contain only lowercase alphanumeric characters, start with a letter or a underscore and with a maximum length of 31 characters"
+        )
+
+
+def create_database(db_name: str, pwd: str):
+    """
+    Create a database and role associated to it
+    Args :
+    name - database name (it will be used also for the role name)
+    pwd  - password used by the created role to connect to the db
+    """
+    credentials = get_db_server_credentials()
+
+    role = credentials["role"]
+    password = credentials["password"]
+    host = credentials["host"]
+    port = credentials["port"]
+
+    url = f"postgresql://{role}:{password}@{host}:{port}"
+    validate_db_name(db_name)
+
```

Comment:
```suggestion
    conn = None
```

If you have a problem to connect to the database `conn` is not instantiated and cannot be used in the finally clause